frontend/benefit/applicant/src/components/applications/forms/application/step2/utils/dates.ts

```diff
@@ -13,23 +13,53 @@ export const getMinEndDate = (
   let months = 0;
   let days = 0;
 
-  if (benefitType === BENEFIT_TYPES.EMPLOYMENT) {
+  if (
+    benefitType === BENEFIT_TYPES.EMPLOYMENT ||
+    benefitType === BENEFIT_TYPES.SALARY
+  ) {
     months = 1;
     days = -1;
-  } else if (benefitType === BENEFIT_TYPES.SALARY) {
-    months = 12;
-    days = -1;
   }
   const parsedStartDate = parseDate(startDate);
-  if (!parsedStartDate || !isValid(parsedStartDate)) {
-    return add(APPLICATION_START_DATE, {
+
+  return add(
+    parsedStartDate && isValid(parsedStartDate)
+      ? parsedStartDate
+      : APPLICATION_START_DATE,
+    {
       months,
       days,
-    });
+    }
+  );
+};
+
+export const getMaxEndDate = (
+  startDate: string | undefined,
+  benefitType: BENEFIT_TYPES | undefined | ''
+): Date | undefined => {
+  if (!benefitType || benefitType === BENEFIT_TYPES.COMMISSION) {
+    return undefined;
   }
 
-  return add(parsedStartDate, {
-    months,
-    days,
-  });
+  let months = 0;
+  let days = 0;
+
+  if (
+    benefitType === BENEFIT_TYPES.EMPLOYMENT ||
+    benefitType === BENEFIT_TYPES.SALARY
+  ) {
+    months = 12;
```

Comment:
I think that the code would be more readable with date-fns arithmetic functions, like:
` return addYears(subDays(date,1),1)`
Even cleaner solutions would ne date-fns/fp and flow: https://github.com/date-fns/date-fns/issues/1532#issuecomment-599940816

Reply:
I followed your suggestion of using the arithmetic functions for more readability.